src/common/utility.ts

```diff
@@ -459,6 +475,42 @@ export class Utility {
         return deployment;
     }
 
+    public static serializeCreateOptions(settings: any, createOptions: any): any {
+        let optionStr: string;
+        if (typeof createOptions === "string") {
+            optionStr = createOptions;
+        } else {
+            optionStr = JSON.stringify(createOptions);
+        }
+        const re = new RegExp(`.{1,${Constants.TwinValueMaxSize}}`, "g");
+        const options = optionStr.match(re);
+        if (options.length > Constants.TwinValueMaxChunks) {
+            throw new Error("Size of createOptions too big. The maxium size of createOptions is 4K");
+        }
+        options.map((value, index) => {
+            if (index === 0) {
+                settings.createOptions = value;
+            } else {
+                const formattedNumber = (`0${index}`).slice(-2);
+                settings[`createOptions${formattedNumber}`] = value;
+            }
+        });
+
+        return settings;
+    }
+
+    private static serializeCreateOptionsForEachModule(modules: any): any {
+        for (const key in modules) {
+            if (modules.hasOwnProperty(key)) {
+                const moduleVar = modules[key];
+                if (moduleVar.settings && moduleVar.settings.createOptions) {
+                    moduleVar.settings = Utility.serializeCreateOptions(moduleVar.settings, moduleVar.settings.createOptions);

```

Comment:
I think we also need to support `createOptionsXX` in deployment.template.json.

Reply:
It's supported